cfgen/cfgen

```diff
@@ -813,8 +823,18 @@ def _infinite_counter(start: int = 0) -> Iterator[int]:
 fidk_gen = _infinite_counter()  # fid key generator
 
 
+def __allocate_base_fidkey(objt: ObjT, fidk: int) -> int:
```

Comment:
In this patch :
As a function header, would be nice if we explain logic about how it works overall.
And comments in function/added code, explaining logic built in steps.

Reply:
Sure, will add.

Reply:
done